### ISSUE TITLE
Add ability to ingest compound, book, and newspaper issue objects

### ISSUE THREAD

**Issue by mjordan** [mjordan/islandora_rest_ingester#9]
For each of these complex objects, the content model passed in `--cmodel` can determine how to assemble the object in each input subdirectory. Input data would include the parts of each child object, like this:

```
input/
├── compound_object_1
│   ├── child_1
│   ├── child_2
│   └── child_3
├── compound_object_2
│   ├── child_1
│   └── child_2
└── compound_object_3
    ├── child_1
    ├── child_2
    ├── child_3
    └── child_4
```

Objects would need to created in a top-down traversal, since to ingest each child we need the PID of its parent.

**Comment by mjordan** [comment]
We should break our the `ingest_object()` function into a class per content model.

**Comment by mjordan** [comment]
We now have an ingester for books. However, derivatives for book pages are not being generated. Investigating.

**Comment by mjordan** [comment]
We now have a compound object ingester. Derivatives also now work for book pages and compound child objects.

**Comment by mjordan** [comment]
Input for books:

```
bookstestinput/
├── 1
│   ├── 1
│   │   └── OBJ.tiff
│   ├── 2
│   │   └── OBJ.tiff
│   ├── 3
│   │   └── OBJ.tiff
│   ├── 4
│   │   └── OBJ.tiff
│   └── MODS.xml
└── 2
    ├── 1
    │   └── OBJ.tiff
    ├── 2
    │   └── OBJ.tiff
    ├── 3
    │   └── OBJ.tiff
    ├── 4
    │   └── OBJ.tiff
    └── MODS.xml
```

For compounds:

```
cpdtestinput/
├── 1
│   ├── 1
│   │   ├── MODS.xml
│   │   └── OBJ.jpg
│   ├── 2
│   │   ├── MODS.xml
│   │   └── OBJ.jpg
│   └── MODS.xml
└── 2
    ├── 1
    │   ├── MODS.xml
    │   └── OBJ.tif
    ├── 2
    │   ├── MODS.xml
    │   └── OBJ.tif
    └── MODS.xml
```

**Comment by mjordan** [comment]
Closed with 6887205eb87fa2c952912c24b7015ea7efe1f4f2.
